ahcore/callbacks/milvus_vector_db_callback.py

```diff
@@ -0,0 +1,111 @@
+from logging import getLogger
+from typing import Any, Callable, Dict
+
+from pymilvus import Collection, MilvusException, connections
+from pytorch_lightning import Callback, LightningModule, Trainer
+
+from ahcore.utils.vector_database.factories import (
+    CollectionType,
+    MilvusCollectionFactory,
+    PrepareDataFactory,
+    PrepareDataTypeDebug,
+    PrepareDataTypePathCLS,
+)
+
+logger = getLogger(__name__)
+
+
+class MilvusVectorDBCallback(Callback):
+    def __init__(
+        self,
+        host: str,
+        port: str,
+        user: str,
+        password: str,
+        collection_name: str,
+        collection_type: str,
+        embedding_dim: int,
+        database: str = "default",
+    ):
+        """
+        Callback to send outputs to Milvus vector database.
+
+        Parameters
+        ----------
+        host : str
+            The host of the Milvus server.
+        port : str
+            The port of the Milvus server.
+        user : str
+            The user to connect to the Milvus server.
+        password : str
+            The password for the user.
+        collection_name : str
+            The name of the collection to insert data into.
+        collection_type : str
+            The type of the collection to insert data into, this will invoke a CollectionType, with specified types
+            and data preparation. E.g. "path_fomo_cls" for a pathology fomo with Image metadata and CLS embeddings.
+        embedding_dim: int
+            The output dimension of your embedding model
+        database : str
+            The database type to connect to. Default is "default".
+        """
+        self.host = host
+        self.port = int(port)
+        self.user = user
+        self.database = database
+        self._password = password
+        self._connection_alias = "ahcore_milvus_vector_db"
+        connections.connect(
+            host=self.host, port=self.port, user=self.user, password=self._password, alias=self._connection_alias
+        )
```

Comment:
Can't this be done with a context manager? E.g. only open it once it's used.

Reply:
Unfortunately not as connections.connect is a statement.

Reply:
You can convert it probably to a context manager, no? Maybe not for this time.